packages/mllp-server/src/utils.ts

```
@@ -56,3 +60,37 @@ export function unpackPidField(pid: string | undefined) {
 function reformUuid(shortId: string) {
   return unpackUuid(crypto.unscramble(shortId));
 }
+
+/**
+ * Extract clean IP address from IPv4-mapped IPv6 address
+ * Removes the ::ffff: prefix if present
+ */
+export function getCleanIpAddress(address: string | undefined): string {
+  if (!address) return "unknown";
+
+  // Remove IPv4-mapped IPv6 prefix if present
+  if (address.startsWith("::ffff:")) {
+    return address.substring(7);
+  }
+
+  return address;
+}
+
+export function lookupHieTzEntryForIp(hieTimezoneDictionary: HieTimezoneDictionary, ip: string) {
+  const cidrToTimezoneRows = Object.entries(hieTimezoneDictionary).map(
+    ([hieName, { cidrBlock, timezone }]) => ({ hieName, cidrBlock, timezone })
+  );
+  const hieRow = cidrToTimezoneRows.find(({ cidrBlock }) => isIpInRange(cidrBlock, ip));
+  if (!hieRow) {
+    throw new MetriportError(`Sender IP not found in any CIDR block`, {
```

do you want to throw here? We don't throw if there's no IP (return "unknown") which makes it seem like we want the system to still process the message...

---

I think we want to throw here and actually not return unknown above, changed my opinion midway through the PR. addressed